### ISSUE TITLE
Remove `preview` from search events

### ISSUE THREAD

**Issue by camdencheek** [sourcegraph/sourcegraph#23982]
See [here](https://github.com/sourcegraph/sourcegraph/pull/23918#pullrequestreview-730613337) for context. 

We don't use preview in the frontend, and relying on it has caused problems in the past. We can significanly reduce payload sizes and memory usage across the application by only sending highlighting offsets, and since we fetch the full file contents for highlighting anyway, we will already have the information at some point to do any post-processing.

**Comment by rvantonder** [comment]
I'm on board with this. One thing I do wonder about:

> and since we fetch the full file contents for highlighting anyway

If we imagined being able to avoid the cost of processing full file contents for highlighting, then I could see sending a "preview" value along does make sense, since we sort of have it at hand while doing matching. I think though, the reason we stopped relying on preview is that that we're not able to achieve accurate highlighting without processing the full file, but I'm not sure. But yeah I do wonder about the potentially great optimization where we don't have to process the whole file for highlighting the subset of "preview" search results. I don't know if precise highlighting means that's a deal breaker. Anyway, since processing the whole file is our default right now, that doesn't block us from removing `preview`.
